selfservice/strategy/oidc/provider_vk.go

```diff
@@ -34,38 +39,56 @@ func NewProviderVK(
 	}
 }
 
-func (g *ProviderVK) Config() *Configuration {
-	return g.config
+func (p *ProviderVK) Config() *Configuration {
+	return p.config
 }
 
-func (g *ProviderVK) oauth2(ctx context.Context) *oauth2.Config {
-	return &oauth2.Config{
-		ClientID:     g.config.ClientID,
-		ClientSecret: g.config.ClientSecret,
-		Endpoint: oauth2.Endpoint{
+func (p *ProviderVK) oauth2(ctx context.Context) *oauth2.Config {
+	var endpoint oauth2.Endpoint
+	if p.config.PKCE == "force" {
```

Comment:
Is there a reason / use case for not using PKCE with VK? Ideally, we would use PKCE by default for all providers that support it.

The explicit config flag for PKCE should only be necessary for the generic provider.

Reply:
I can enable PKCE for `pkce == "auto"` too. I thought `auto` should enable PKCE only if `/.well-known/openid-configuration` returns that it's supported, that's why I didn't enable PKCE for `auto`.
I don't know the reason why we shouldn't use PKCE. But enabling non-PKCE configuration when `pkce` is explicitly set to `never` seems logical.